cmd/talosctl/cmd/talos/get.go

```diff
@@ -230,45 +224,61 @@ func aggregateEvents(ctx context.Context, outCh chan<- nodeAndEvent, watchCh <-c
 	}
 }
 
-// completeResource represents tab complete options for `get` and `get *` commands.
-//
-//nolint:gocyclo
-func completeResource(resourceType string, hasAliases bool, completeDot bool) []string {
-	var resourceOptions []string
+// completeResourceDefinition represents tab complete options for `get` and `get *` commands.
+func completeResourceDefinition(withAliases bool) ([]string, cobra.ShellCompDirective) {
+	var result []string
 
-	callbackResource := func(_ context.Context, _ string, r resource.Resource, callError error) error {
-		if callError != nil {
-			return callError
+	if WithClientNoNodes(func(ctx context.Context, c *client.Client) error {
+		items, err := safe.StateList[*meta.ResourceDefinition](ctx, c.COSI, resource.NewMetadata(meta.NamespaceName, meta.ResourceDefinitionType, "", resource.VersionUndefined))
+		if err != nil {
+			return err
 		}
 
-		if completeDot {
-			resourceOptions = append(resourceOptions, r.Metadata().ID())
+		for iter := safe.IteratorFromList(items); iter.Next(); {
+			if withAliases {
+				result = append(result, iter.Value().TypedSpec().Aliases...)
+			}
+
+			result = append(result, iter.Value().Metadata().ID())
 		}
 
-		if !hasAliases {
-			return nil
+		return nil
+	}) != nil {
+		return nil, cobra.ShellCompDirectiveError
```

Comment:
Tested it, works great.... 
Minor nit, please add `| cobra.ShellCompDirectiveNoFileComp` here. In the event of an error (for example a timeout), it should not autocomplete with files